cypress/e2e/functional/tile_tests/geometry_tool_spec.js

```diff
@@ -277,6 +356,164 @@ context('Geometry Tool', function () {
+    clueCanvas.clickToolbarButton('geometry', 'polygon');
+    geometryToolTile.getGraphPoint().should("have.length", 3);
+    geometryToolTile.getGraphPoint().eq(0).click();
+    geometryToolTile.getGraphPoint().eq(1).click();
+    geometryToolTile.getGraphPoint().eq(2).click();
+    geometryToolTile.getGraphPoint().eq(0).click();
+    geometryToolTile.getGraphPolygon().should("have.length", 1);
+    geometryToolTile.getGraphPoint().should("have.length", 3);
+
+    // Add a point to the existing polygon
+    cy.log('Add a point to the existing polygon');
+    geometryToolTile.clickGraphPosition(10, 0); // Reuse existing point
+    geometryToolTile.clickGraphPosition(15, 5);
+    geometryToolTile.clickGraphPosition(5, 5); // Reuse existing point
+    clueCanvas.clickToolbarButton('geometry', 'polygon');
+    // check number of points
+    geometryToolTile.getGraphPoint().should("have.length", 4);
+
+    // Verify the polygon count is still 1
+    geometryToolTile.getGraphPolygon().should("have.length", 1);
+
+    // Create a second polygon that shares the same points as the first
+    cy.log('Create a second polygon that shares the same points as the first');
+    clueCanvas.clickToolbarButton('geometry', 'polygon');
+    geometryToolTile.clickGraphPosition(15, 10); // new point
+    geometryToolTile.clickGraphPosition(15, 5); // shared point
+    geometryToolTile.clickGraphPosition(20, 5); // new point
+    geometryToolTile.clickGraphPosition(15, 10); // close the polygon
+
+    // Point should be shared
+    geometryToolTile.getGraphPoint().should("have.length", 6); // New point added
+
+    // Store the original coordinates for comparison
+    let originalCx, originalCy;
+    clueCanvas.clickToolbarButton('geometry', 'select');
+    geometryToolTile.clickGraphPosition(15, 5); // shared point
+    geometryToolTile.getSelectedGraphPoint().then(($point) => {
+      originalCx = parseFloat($point.attr('cx'));
+      originalCy = parseFloat($point.attr('cy'));
+      cy.wrap(originalCx).as('originalCx');
+      cy.wrap(originalCy).as('originalCy');
+    });
+
+    // Add an angle label
+    geometryToolTile.getGraphPointLabel().contains('°').should('not.exist');
+    clueCanvas.clickToolbarButton('geometry', 'select');
+    geometryToolTile.selectGraphPoint(20, 5); // this point is a 45-degree angle but sometimes it starts at 44
+    clueCanvas.clickToolbarButton('geometry', 'label');
+    geometryToolTile.toggleAngleCheckbox();
+
+    // Use a flexible check to verify either '44°' or '45°'
+    cy.log('Verify the initial angle is either 44° or 45°');
+    geometryToolTile.getGraphPointLabel().then(($label) => {
+      const angleText = $label.text();
+      if (angleText.includes('44°')) {
+        cy.wrap(44).as('initialAngle');
+      } else if (angleText.includes('45°')) {
+        cy.wrap(45).as('initialAngle');
+      } else {
+        throw new Error(`Unexpected angle: ${angleText}`);
+      }
+    });
```

Comment:
It seems odd to me that the angle is not consistent, but this is a good solution if so. And I guess it's not worth spending time investigating such a small inconsistency.